community/consistency-check/src/main/java/org/neo4j/consistency/checking/SchemaRecordCheck.java

```diff
@@ -179,12 +170,26 @@ else if ( phase == Phase.CHECK_OBLIGATIONS )
         }
     }
 
-    private void checkMandatoryProperty( int propertyKey, RecordAccess records,
+    private void checkMandatoryNodePropertyRule( MandatoryNodePropertyConstraintRule rule, DynamicRecord record,
+            RecordAccess records, CheckerEngine<DynamicRecord,ConsistencyReport.SchemaConsistencyReport> engine )
+    {
+        if ( phase == Phase.CHECK_RULES )
```

Comment:
These methods might indeed be part of `Phase` enum. The only drawback that I see is size of the enum and some methods with empty implementations.
Do not think it is possible to move this check out because it participates in `if-else-if` clause.
Will investigate the possibility of making this prettier.